sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client_helpers.py

```diff
@@ -0,0 +1,1351 @@
+    serialize_blob_tags,
+    serialize_query_format, get_access_conditions
+)
+from ._deserialize import deserialize_blob_stream
+
+from ._encryption import modify_user_agent_for_encryption
+from ._models import (
+    BlobType,
+    BlobBlock,
+    BlobProperties,
+    QuickQueryDialect,
+    DelimitedJsonDialect,
+    DelimitedTextDialect,
+)
+from ._upload_helpers import _any_conditions
+
+if TYPE_CHECKING:
+    from urllib.parse import ParseResult
+
+_ERROR_UNSUPPORTED_METHOD_FOR_ENCRYPTION = (
+    'The require_encryption flag is set, but encryption is not supported'
+    ' for this method.')
+
+
+def _parse_url(
+    account_url: str,
+    container_name: str,
+    blob_name: str
+) -> Tuple["ParseResult", Any]:
+    """Performs initial input validation and returns the parsed URL, SAS token, and path snapshot.
+
+    :param str account_url: The URL to the storage account.
+    :param str container_name: The name of the container.
+    :param str blob_name: The name of the blob.
+    :returns: The parsed URL, SAS token, and path snapshot.
+    :rtype: Tuple[ParseResult, Any]
+    """
+    try:
+        if not account_url.lower().startswith('http'):
+            account_url = "https://" + account_url
+    except AttributeError as exc:
+        raise ValueError("Account URL must be a string.") from exc
+    parsed_url = urlparse(account_url.rstrip('/'))
+
+    if not (container_name and blob_name):
+        raise ValueError("Please specify a container name and blob name.")
+    if not parsed_url.netloc:
+        raise ValueError(f"Invalid URL: {account_url}")
+
+    path_snapshot, sas_token = parse_query(parsed_url.query)
+
+    return parsed_url, sas_token, path_snapshot
+
+def _format_url(container_name, scheme, blob_name, query_str, hostname):
+    container_name = container_name
+    if isinstance(container_name, str):
+        container_name = container_name.encode('UTF-8')
+    return f"{scheme}://{hostname}/{quote(container_name)}/{quote(blob_name, safe='~/')}{query_str}"
+
+def _encode_source_url(source_url):
+    parsed_source_url = urlparse(source_url)
+    source_scheme = parsed_source_url.scheme
+    source_hostname = parsed_source_url.netloc.rstrip('/')
+    source_path = unquote(parsed_source_url.path)
+    source_query = parsed_source_url.query
+    result = [f"{source_scheme}://{source_hostname}{quote(source_path, safe='~/')}"]
+    if source_query:
+        result.append(source_query)
+    return '?'.join(result)
+
+def _upload_blob_options(  # pylint:disable=too-many-statements
+    require_encryption,
+    encryption_version,
+    key_encryption_key,
+    key_resolver_function,
+    scheme,
+    config,
+    sdk_moniker,
+    client,
+    data: Union[bytes, str, Iterable[AnyStr], AsyncIterable[AnyStr], IO[AnyStr]],
+    blob_type: Union[str, BlobType] = BlobType.BlockBlob,
+    length: Optional[int] = None,
+    metadata: Optional[Dict[str, str]] = None,
+    **kwargs
+    ) -> Dict[str, Any]:
```

Comment:
I see a bit of difference in formatting for these function defs in here. Let's get them all consistent to something like:

```py
def function_name(
    param1,
    param2,
    kwargs
) -> ReturnType:
```